setup.py

```diff
@@ -49,22 +50,24 @@ def _hash(filepath):
         file_hash = base64.b64encode(hashlib.sha512(file_as_bytes).digest())
         return "sha512-" + file_hash.decode()
 
-    os.makedirs(external_dir_single, exist_ok=True)
+    external_dir_single.mkdir(exist_ok=True)
     # Used for downloading e.g. Leaflet single file
     if out_name is None:
         out_name = url.split("/")[-1]
-    dst_path = os.path.join(external_dir_single, out_name)
-    if os.path.isfile(dst_path):
-        if integrity_hash and (_hash(dst_path) == integrity_hash):
-            return
-        else:
-            return
+    dst_path = external_dir_single / out_name
+
+    if dst_path.is_file():
+        return
+    if integrity_hash and (_hash(str(dst_path)) != integrity_hash):
```

Comment:
@wang-boyu Thank you!  We think this is the logic you're looking for (note that we changed == to !=); the setup should stop if the hash is provided and does not match.